test/e2e/specs/customer_admin.go

```diff
@@ -81,6 +81,50 @@ var _ = Describe("Openshift on Azure customer-admin e2e tests [CustomerAdmin][Fa
 		Expect(err).ToNot(HaveOccurred())
 	})
 
+	It("should be able to edit project-request", func() {
+		projectTemplate, err := sanity.Checker.Client.CustomerAdmin.TemplateV1.Templates("openshift").Get("project-request", metav1.GetOptions{})
+		Expect(err).ToNot(HaveOccurred())
+		// modify project template object
+		projectTemplate.SetAnnotations(map[string]string{
+			"foo.annotation.com": "bar",
+		})
+		_, err = sanity.Checker.Client.CustomerAdmin.TemplateV1.Templates("openshift").Update(projectTemplate)
+		Expect(err).ToNot(HaveOccurred())
+	})
+
+	It("should be able to edit self-provisioners clusterrolebinding", func() {
+		crbOriginal, err := sanity.Checker.Client.CustomerAdmin.RbacV1.ClusterRoleBindings().Get("self-provisioners", metav1.GetOptions{})
+		Expect(err).ToNot(HaveOccurred())
+		crb := crbOriginal.DeepCopy()
+
+		// create ClusterRoleBinding which would allow us to test edit
+		// and would not break any other tests
+		subject := []rbacv1.Subject{
+			{
+				Name: "system:authenticated:oauth",
+				Kind: "Group",
+			},
+			{
+				Name: "osa-customer-admins",
+				Kind: "Group",
+			},
+		}
+
+		crb.Subjects = subject
+
+		_, err = sanity.Checker.Client.CustomerAdmin.RbacV1.ClusterRoleBindings().Update(crb)
+		Expect(err).ToNot(HaveOccurred())
+
+		// HACK: openshift reconciler should not touch this role because of:
+		// openshift.io/reconcile-protect: "true"
+		// we need to confirm this
+		time.Sleep(5 * time.Second)
+
+		crbUpdated, err := sanity.Checker.Client.CustomerAdmin.RbacV1.ClusterRoleBindings().Get("self-provisioners", metav1.GetOptions{})
+		Expect(err).ToNot(HaveOccurred())
+		Expect(len(crbUpdated.Subjects)).To(Equal(len(crb.Subjects)))
```

Comment:
Do we want to check them one by one for equality? We didn't actually demonstrate these lengths weren't equal before, so it's possible this test is passing by coincidence.

Reply:
This makes sure this flow works:
	// Flow:
	// 1. Read rolebinding with default system:authenticated:oauth value (single value by default)
	// 2. Update it with 2 values (group + system:authenticated:oauth) and our reconcile annotation
	// 3. Commit to API
	// 4. Confirm what we committed is still in api/etcd